### PR TITLE
kind: minor fixes, documentation, ...

### PR DESCRIPTION
**Pull request by BenTheElder**
I cleaned up the documentation a bit, I'm still filtering things into markdown and trying to better organize things etc..
Also set up OWNERS and fixed some minor nits with `kind/hack/generate.sh`. This generally contains some things I fixed up this weekend while working on a more major PR, I'd like to include them separately to continue to keep the PR sizes down :-)

### CODE REVIEW

File: kind/docs/design.md [hunk 1]
```diff
@@ -5,35 +5,68 @@ This is the root design documentation for `kind`. See also the project
...
-- we do a few tricks to minimize unnecessary services and inform systemd that it
-is in docker. (see the [Dockerfile](./../images/node/Dockerfile))
+We initially offload this type of state into the containers and Docker. 
+Similarly the container names are automatically managed by `kind`, though
+we will select over labels instead of names because these are less brittle and
+are properly namespaced. Doing this also avoids us needing to manage anything
+on the host filesystem, but should not degrade usage.
 
-# TODO(bentheelder): fill in further docs as things stabilize / functionality is checked in, particularly around node management
+The `KUBECONIFG` will be bind-mounted to a temp directory, with the tooling
```
Comment (krzyzacy) [inline kind/docs/design.md]: `KUBECONFIG`? :-)

Comment (BenTheElder) [reply kind/docs/design.md]: thanks, fixed

**Comment by BenTheElder** [comment]
fixed the spelling lint 😬

**Pull request merged**